src/irmin-pack/store_intf.ml

```diff
@@ -45,4 +45,29 @@ module type Store = sig
   val migrate : Irmin.config -> unit
 
   exception Unsupported_version of IO.version
+
+  module Checks (Index : Pack_index.S) : sig
+    val integrity_check :
+      ?ppf:Format.formatter ->
+      auto_repair:bool ->
+      check_contents:
+        (offset:int64 ->
+        length:int ->
+        Index.key ->
+        (unit, [ `Absent_value | `Wrong_hash ]) result) ->
+      check_nodes:
+        (offset:int64 ->
+        length:int ->
+        Index.key ->
+        (unit, [ `Absent_value | `Wrong_hash ]) result) ->
+      check_commits:
+        (offset:int64 ->
+        length:int ->
+        Index.key ->
+        (unit, [ `Absent_value | `Wrong_hash ]) result) ->
```

Comment:
```suggestion
        ((offset:int64 ->
        length:int ->
        Index.key ->
        (unit, [ `Absent_value | `Wrong_hash ]) result) as 'check) ->
      check_nodes:'check ->
      check_commits:'check ->
```

(or could make it a single ``check : kind:[ `Contents | `Node | `Commit ] -> ...`` argument, which I think would be cleaner at the call-site.)

Reply:
Yes, thanks!